src/WebJobs.Script/Description/Rpc/WorkerLanguageInvoker.cs

```diff
@@ -70,13 +76,28 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             };
 
             ScriptInvocationResult result;
-            _invocationBuffer.Post(invocationContext);
+            _logger.LogDebug($"Sending invocation id:{invocationId}");
+            await _functionDispatcher.InvokeAsync(invocationContext);
             result = await invocationContext.ResultSource.Task;
 
             await BindOutputsAsync(triggerValue, context.Binder, result);
             return result.Return;
         }
 
+        private async Task DelayUntilFunctionDispatcherInitialized()
+        {
+            // Need to wait for atleast one language worker process to be initialized before setting ScriptHostState to running
+            // for non dotnet functions
+            if (_functionDispatcher != null && _functionDispatcher.State == FunctionDispatcherState.Initializing)
+            {
```

Comment:
@fabiocav - Moved the logic that checks for function dispatcher start to WorkerLanguageInvoker. Can you please take a final look?